packages/design-system/src/components/Text/Text.tsx

```diff
@@ -0,0 +1,24 @@
+import { clsx } from 'clsx';
+import { forwardRef } from 'react';
+import { textRecipe, TextRecipe } from './Text.css';
+
+type TextProps<T extends keyof JSX.IntrinsicElements> = TextRecipe & {
+  as?: T;
+  className?: string;
+  children: React.ReactNode;
+  dataTestId?: string;
+};
+
+export const Text = forwardRef<HTMLElement, React.PropsWithChildren<TextProps<any>>>(
+  ({ variant, weight, children, color, dataTestId, className, as: Tag = 'p' }, forwardedRef) => {
+    return (
+      <Tag
+        className={clsx(textRecipe({ variant, weight, color }), className)}
+        ref={forwardedRef}
+        data-testid={dataTestId}
+      >
+        {children}
+      </Tag>
+    );
+  }
+);
```

Comment:
I would avoid reinventing the data-testid convention, even if it is less pleasant to write in the type definition 😁

```suggestion
type TextProps<T extends keyof JSX.IntrinsicElements> = TextRecipe & {
  as?: T;
  className?: string;
  children: React.ReactNode;
 'data-testid'?: string;
};

export const Text = forwardRef<HTMLElement, React.PropsWithChildren<TextProps<any>>>(
  ({ variant, weight, children, color, 'data-testid': dataTestId, className, as: Tag = 'p' }, forwardedRef) => {
    return (
      <Tag
        className={clsx(textRecipe({ variant, weight, color }), className)}
        ref={forwardedRef}
        data-testid={dataTestId}
      >
        {children}
      </Tag>
    );
  }
);
```